### PR TITLE
[CHK-3193] feat: add TransactionInfo data structure into DeadLetterEvent

### PR DESCRIPTION
**Pull request by scaminati-bv**
#### List of Changes
- add TransactionInfo class
- create TransactionInfoDetailsData interface for gateway custom fields
- add transactionInfo property into DeadLetterEvent class

#### Motivation and Context
This PR add transactionInfo property into DeadLetterEvent document. This modification is needed by helpdesk for refund operations.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by scaminati-bv** [comment]
@infantesimone I mark this PR ready for review.
The most important thing to check is this NotNull properties in [TransactionInfo class](https://github.com/pagopa/pagopa-ecommerce-commons/pull/217/files#diff-9a621d8ace35cc47350870b783adb53333e5fac9cf9fc681c15fd2f6a5a3221b)

File: src/main/java/it/pagopa/ecommerce/commons/documents/v2/info/TransactionInfo.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    @NotNull
+    private String authorizationRequestId;
+
+    @NotNull
+    private TransactionStatusDto eCommerceStatus;
+
+    @NotNull
+    private TransactionAuthorizationRequestData.PaymentGateway gateway;
+
+    private List<String> paymentTokens;
```
Comment (infantesimone) [inline src/main/java/it/pagopa/ecommerce/commons/documents/v2/info/TransactionInfo.java:34]: you can assume `notNull`

File: src/main/java/it/pagopa/ecommerce/commons/documents/v2/info/TransactionInfo.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    private String pspId;
```
Comment (infantesimone) [inline src/main/java/it/pagopa/ecommerce/commons/documents/v2/info/TransactionInfo.java:36]: you can assume `NotNull`

**Comment by infantesimone** [comment]
models present in `v2/info` are generic or for the deadletter structure? because this impacts which fields we can assume are null.

**Comment by infantesimone** [comment]
please add unit test (simple but useful to check null value)

**Comment by infantesimone** [comment]
I think it is useful to have models dedicated to the deadletter for greater control over the validity of documents in deadletter. What do you think? in this case the name of the classes or the package could refer to the deadletter

**Comment by scaminati-bv** [comment]
> models present in `v2/info` are generic or for the deadletter structure? because this impacts which fields we can assume are null.

At the moment i consider this field generic, so i put this on info package, but it is used only into deadLetterEvent class.

**Comment by scaminati-bv** [comment]
> I think it is useful to have models dedicated to the deadletter for greater control over the validity of documents in deadletter. What do you think? in this case the name of the classes or the package could refer to the deadletter

Yes i agree. Now i change package name to deadletter and change prefix of classes.

File: src/main/java/it/pagopa/ecommerce/commons/documents/v2/info/TransactionInfo.java (hunk 1 above)
Comment (scaminati-bv) [reply src/main/java/it/pagopa/ecommerce/commons/documents/v2/info/TransactionInfo.java:34]: done here 9ef26404c69fe0f32f00ad15b962c9d61009833f

**Comment by scaminati-bv** [comment]
> please add unit test (simple but useful to check null value)

At the moment all unit test do not work

File: src/main/java/it/pagopa/ecommerce/commons/documents/v2/deadletter/DeadLetterTransactionInfo.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    private String pspId;
+
+    private String paymentMethodName;
+
+    private Integer grandTotal;
+
+    private String rrn;
+
+    private DeadLetterTransactionInfoDetailsData details;
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/ecommerce/commons/documents/v2/deadletter/DeadLetterTransactionInfo.java:45]: If these can be nullable please annotate them with `@Nullable` :)

Comment (scaminati-bv) [reply src/main/java/it/pagopa/ecommerce/commons/documents/v2/deadletter/DeadLetterTransactionInfo.java:45]: done here 1da6f0537eae92fe0c961a15c2169d9c2397ff0f

Comment (scaminati-bv) [reply src/main/java/it/pagopa/ecommerce/commons/documents/v2/deadletter/DeadLetterTransactionInfo.java:45]: you are referring to only details?

**Review by infantesimone — APPROVED** [review]
